### DISCUSSION TITLE
Synology: Variables

### DISCUSSION THREAD

**Discussion by mckee14** [felddy/foundryvtt-docker#291] [Ideas]
Dear @felddy 

A year ago, we discussed (in #208) the synology docker GUI and the "problem" that it loads some variables by default (which creates problems when you want to update the image to a new version).

Yesterday I stumbled over another thread about the docker GUI. One guy mentions, **that synology docker GUI exposes variables which are declared as ENV**. If only needed to build the image, one should declare them as ARG.

Personally, I have no idea what that means, I just thought that I report this. 

Best case, we can get rid of these variables in the GUI. Worst case, you just spent 2 minutes of your life reading this :)

Anyway, thank you for your work, it's great that noobs like me can run Foundry out of the box!

**Comment by felddy** [comment]
Thank you for thoughtful message.  

The interaction between `ENV` and `ARG` is subtle, but important.  There are many arguments that are used in the `Dockerfile` to build the container:

https://github.com/felddy/foundryvtt-docker/blob/e08f2273e497349b8dbf2e4a4da272a09e136705/Dockerfile#L54-L64

But there are also a couple of environment variables that need to be passed to the image: `FOUNDRY_HOME` and `FOUNDRY_VERSION`.   There are ways that I could inject the `FOUNDRY_VERSION` without using an environment variable, but I am hesitant to do that right now as I expect there are deployments that are using the environment variable as it is implemented.

It seems like the behavior you are seeing is a long-standing bug with the Synology/Portainer implementation, documented in the following: 
- https://github.com/portainer/portainer/issues/5813
- https://github.com/portainer/portainer/issues/2952

This ticket https://github.com/portainer/portainer/issues/3964 claims it was fix in version 2.0.

There might be a solution to this behavior here: 
https://discourse.pi-hole.net/t/pihole-portainer-docker-5-8-1-startup-error/48722/7

This sounds very frustrating and I wish I could do more to help resolve it from this side, but I don't think I can safely do so.

I'll keep an eye out for any movement on this and update the discussion if I see anything.

Thank you again for letting me know this still affects you.

**Comment by mckee14** [comment]
Update for other Synology Docker users:

1. stop the container and download the latest image. I use the "latest" tag and the old image will be overwritten automatically
2. open the settings for the stopped container. in the extended settings, delete the exposed variables (home, foundry_version etc). You can keep the variables you need (foundry login, pw, admin-pw etc).
3. in the docker gui, select the container and klick Action > reset container.
4. launch container

all in all, you download an image, delete some lines and you're good to go. when you reset the container, the new and correct exposed variables will be loaded and all works fine.
